### ISSUE TITLE
Cell Snippet reactivity

### ISSUE THREAD

**Issue by Tihomir971** [walker-tx/svelte5-tanstack-table-examples#4]
Hi,
I have problem with cell reactivity when using snippet:

This is reacive:
`colHelp.accessor('pricePurchase', { header: 'Purchase' }) ,`

This is not reactive:
```
colHelp.accessor('pricePurchase', {
		header: 'Purchase',
		cell: ({ cell }) => {
			return renderSnippet(rightAlignSnippet, cell.getValue());
		}
	})
```

This is snippet:
```
const rightAlignSnippet = createRawSnippet<[string]>((getValue) => {
	const value = getValue() ?? 0;
	return {
		render: () => `<div class="text-right">${value}</div>`
	};
});
```

**Comment by Tihomir971** [comment]
I find that snippet must be defined inside cell definition like:	
```
	cell: ({ cell }) => {
			const rightAlignSnippet = createRawSnippet<[string | number | null]>((getValue) => {
				const value = getValue();
				return {
					render: () => `<div class="text-right">${value}</div>`
				};
			});
			return renderSnippet(rightAlignSnippet, cell.getValue());
		}
```
This way snippet cant be reusable :(

**Comment by walker-tx** [comment]
Thank you for creating an issue!

Would you please provide a minimal reproduction of this issue via either stack blitz or a GH repo?

**Comment by Tihomir971** [comment]
> Thank you for creating an issue!
> 
> Would you please provide a minimal reproduction of this issue via either stack blitz or a GH repo?

Try on this link:
[Stackblitz](https://stackblitz.com/edit/sveltejs-kit-template-default-im3mzp)

**Comment by walker-tx** [comment]
Ok - thank you for sharing the example! This has been a helpful exercise for me to understand how `createRawSnippet` works.

# The Fix

First, rename the `columns.ts` file to `columns.svelte.ts`. Be sure to update relevant imports in other dependent files.

Next, in `columns.svelte.ts` update your snippet's return object to include a setup method with an `effect` rune that updates the `node` when `getValue()` changes:

```ts
// columns.svelte.ts
const rightAlignSnippet = createRawSnippet<[string | number | null]>(
  (getValue) => {
    const value = getValue();
    return {
      render: () => `<div class="text-right">${value}</div>`,
      setup: (node) => {
        // Add `effect` rune here
        $effect(() => {
          node.textContent = String(getValue());
        })
      }
    };
  }
);
```

# The Why

I could be wrong, but here's my best guess.

Out of the box, `createRawSnippet` is _not_ reactive. In other words, after the initial render, there's nothing that updates the DOM in reaction to prop/param changes. 

Ok - now you're saying: "Wait a second, defining the raw snippet inside the `cell` method of the column's definition was reactive!"

Sort of. The snippet wasn't actually reactive - but the _table_ was. Each time data changed, the table recalculated the cells, running each `columnDef.cell` method, thus creating a _new instance_ of the snippet for each cell. Therefor, the snippet never needed to react - each table render simply replaced your old snippet with the new one. It wasn't a reaction, it was a **_replacement_**!

On the other hand, in your first (non-reactive) post, you created a single instance of `rightAlignSnippet`, and re-used it for each render of `cell`. Each time data was changed, the table re-rendered using the same instance of `rightAlignSnippet`. Svelte doesn't change the DOM contents because the snippet instance from initial render and the instance from subsequent renders are the same - even though their params are different. Therefor, since the cells aren't being replaced with a new instance in this implementation, and since `createRawSnippet` does not come with reactivity out of the box, we must manually instruct the Svelte as to what should happen when a new value is passed.

# TL;DR

Snippets made with `createRawSnippet` need reactivity to be manually implemented.

**Comment by walker-tx** [comment]
Closed by PR #5

**Comment by magiceupho** [comment]
Hi,
Thank you for the workaround, but how can I get the reactivity when I need to render a little bit of html in the snippet please ?
This doesn't work:
```ts
export const rulesSnippet = createRawSnippet<[GameSession]>(
	(gameSession) => {
		let rules = $derived(formatRules(gameSession()));
		return {
			render: () => `<div>${rules.length > 0 ? rules.join('') : '/'}</div>`,
		};
	}
);
```
And if I use the setup function, html is escaped.

Regards,
magiceupho

**Comment by walker-tx** [comment]
@magiceupho, Please provide a minimal reproduction of your issue in StackBlitz or as a GH repo.
